services/convert/user.go

```diff
@@ -51,7 +51,7 @@ func toUser(ctx context.Context, user *user_model.User, signed, authed bool) *ap
 		ID:          user.ID,
 		UserName:    user.Name,
 		FullName:    user.FullName,
-		Email:       user.GetEmail(),
+		Email:       user.GetPlaceholderEmail(),
```

Comment:
So we need to check if signed is true?

Reply:
The file already cntians this code down below which does the check:
```go
// hide primary email if API caller is anonymous or user keep email private
if signed && (!user.KeepEmailPrivate || authed) {
	result.Email = user.Email
}
```
As `user.GetMail()` only checks if the Mail is private, but not if someone is logged in, it always returns the Mail if it is not private, so this Check hasn't worked. Now that `Email` is set to the Placeholder before this Check, it is now working.